### ISSUE TITLE
The type initializer for 'Xamarin.Forms.DataGrid.DataGrid' threw an exception.

### ISSUE THREAD

**Issue by jagbarcelo** [akgulebubekir/Xamarin.Forms.DataGrid#50]
Anyone else is seeing this? The whole exception contents are:

```
   System.TypeInitializationException: The type initializer for 'Xamarin.Forms.DataGrid.DataGrid' threw an exception. ---> System.Reflection.TargetException: Exception of type 'System.Reflection.TargetException' was thrown.
   at System.Reflection.RuntimeMethodInfo.ThrowNoInvokeException()
   at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Xamarin.Forms.ImageSource.FromResource(String resource, Assembly sourceAssembly)
   at Xamarin.Forms.DataGrid.DataGrid..cctor()
   --- End of inner exception stack trace ---
   at Xamarin.Forms.DataGrid.DataGrid.get_HeaderBackground()
```

Digging deeper in the InnerException, it shows:

```
   System.Reflection.TargetException: Exception of type 'System.Reflection.TargetException' was thrown.
   at System.Reflection.RuntimeMethodInfo.ThrowNoInvokeException()
   at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Xamarin.Forms.ImageSource.FromResource(String resource, Assembly sourceAssembly)
   at Xamarin.Forms.DataGrid.DataGrid..cctor()
```

I have even downloaded a copy of Xamarin.Forms.DataGrid and placed it alongside the rest of my projects, in my solution (instead of using the NuGet version), so that I could trace into it. I have confirmed that the exception is raised the getter of HeaderBackground in DataGrid.xaml.cs, line:192
```
public Color HeaderBackground
{
	get { return (Color)GetValue(HeaderBackgroundProperty); }
	set { SetValue(HeaderBackgroundProperty, value); }
}
```

However, it seems it is not DataGrid's fault, but Xamarin's. An important note, it only happens in UWP projects (Android works fine and I have no way to test iOS platform).

Can someone confirm this issue? If so, a comment in [Is ImageSource.FromResource() working for UWP ?](https://forums.xamarin.com/discussion/106554/is-imagesource-fromresource-working-for-uwp) would be much appreciated.

**Comment by akgulebubekir** [comment]
Hi,

It seems you're not using same version of Xamarin.Forms on all your project. Please be sure you're using same version of it.

**Comment by jagbarcelo** [comment]
No, I already double-checked that. We were using Xamarin.Forms 2.4.0.74863 everywhere (I updated your project when added it to my solution, since it was using Xamaring.Forms 2.4.0.38779). I even updated everything to the latest (back then) Xamarin.Forms 2.5.0.77107 released 3 days ago. Today, I'm trying with Xamarin.Forms 2.5.0.91635 (released 9 hours ago) with the same results too.

I've been struggling with this for a week now and it seems it is Xamarin's fault not yours. Can you please confirm if you can reproduce this error? It only fails in UWP. Thanks.

**Comment by jagbarcelo** [comment]
In order to make minimal repro, I have just created an empty new **Cross Platform App (Xamarin)** project, added the reference to Xamarin.Forms.DataGrid package through NuGet, updated any other package so that everything is in its most recent version and used these files:

**App.xaml**
```
<?xml version="1.0" encoding="utf-8" ?>
<Application xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="ErrorTest.App">
    <Application.Resources>
        <ResourceDictionary>
            
            <Color x:Key="PrimaryColor">#FF9100</Color>

        </ResourceDictionary>
    </Application.Resources>
</Application>
```
**MainPage.xaml**
```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:local="clr-namespace:ErrorTest"
             xmlns:dg="clr-namespace:Xamarin.Forms.DataGrid;assembly=Xamarin.Forms.DataGrid"
             x:Class="ErrorTest.MainPage">

    <ContentPage.Content>
        <dg:DataGrid ItemsSource="{Binding Log}" RowHeight="22" HeaderHeight="40" BorderColor="#CCCCCC" HeaderBackground="{StaticResource PrimaryColor}" ActiveRowColor="{StaticResource PrimaryColor}" >

            <dg:DataGrid.HeaderFontSize>
                <OnIdiom x:TypeArguments="x:Double">
                    <OnIdiom.Tablet>15</OnIdiom.Tablet>
                    <OnIdiom.Phone>12</OnIdiom.Phone>
                </OnIdiom>
            </dg:DataGrid.HeaderFontSize>

            <dg:DataGrid.Columns>
                <dg:DataGridColumn Title="DateTime" StringFormat="{}{0:dd/MM/yyyy HH:mm:ss}" PropertyName="DateTime" Width="*" VerticalContentAlignment="Start"/>
                <dg:DataGridColumn Title="Severity" PropertyName="Severity" Width="*"/>
                <dg:DataGridColumn Title="Message" PropertyName="Message" Width="1.5*" VerticalContentAlignment="Start"/>
                <dg:DataGridColumn Title="MessageParam" PropertyName="MessageParam" Width="1.5*"/>
            </dg:DataGrid.Columns>
        </dg:DataGrid>
    </ContentPage.Content>
       
</ContentPage>
```

It fails in UWP and runs fine in Android. I am using Windows 10 (Version 1709) and Visual Studio Community 2017 (Version 15.4.4) The exception is thrown inside Xamarin.Forms code (not yours) but maybe you could catch it somehow in your code and handle it somehow so that at least the application does not crash.

**Comment by akgulebubekir** [comment]
Hi,

I've reproduced the problem with Xamarin.Forms.2.4.0.38779, when I downgraded to 2.3.4.231 it fixed somehow. I'l investigate that issue further.

**Comment by jagbarcelo** [comment]
I've submitted a bug to Xamarin team. For more information and a minimum repro case (without your component involved), check https://bugzilla.xamarin.com/show_bug.cgi?id=60731

**Comment by akgulebubekir** [comment]
Workaround : Change your region to United States

**Comment by akgulebubekir** [comment]
Fixed with [1.9.9](https://www.nuget.org/packages/Xamarin.Forms.DataGrid/1.9.9)
